File: reimi/frontend/reimi_app/lib/domain/value_objects/wind_direction.dart
import 'package:json_annotation/json_annotation.dart';

enum WindDirection {
  @JsonValue('NORTH')
  north,
  @JsonValue('NORTH_EAST')
  northEast,
  @JsonValue('EAST')
  east,
  @JsonValue('SOUTH_EAST')
  southEast,
  @JsonValue('SOUTH')
  south,
  @JsonValue('SOUTH_WEST')
  southWest,
  @JsonValue('WEST')
  west,
  @JsonValue('NORTH_WEST')
  northWest,
  @JsonValue('UNKNOWN')
  unknown,
}

File: reimi/backend/reimi_app/src/main/java/com/reimi/reimi_app/domain/model/report/value/WindDirection.java
package com.reimi.reimi_app.domain.model.report.value;

public enum WindDirection {
    NORTH("北"),
    NORTH_EAST("北東"),
    EAST("東"),
    SOUTH_EAST("南東"),
    SOUTH("南"),
    SOUTH_WEST("南西"),
    WEST("西"),
    NORTH_WEST("北西"),
    UNKNOWN("分からない");

    private final String label;

    WindDirection(String label) {
        this.label = label;
    }

    public String getLabel() {
        return label;
    }
}
